core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -138,6 +169,22 @@ public boolean accept(BytesRef value) {
 
     }
 
+    class PartitionedOrdinalsFilter extends OrdinalsFilter {
+        @Override
+        public LongBitSet acceptedGlobalOrdinals(RandomAccessOrds globalOrdinals) throws IOException {
+            final long numOrds=globalOrdinals.getValueCount();
+
+            LongBitSet acceptedGlobalOrdinals = new LongBitSet(numOrds);
+            final int zeroBasedPartition = incPartition-1;
+            for (int i = 0; i < numOrds; i++) {
+                if (i % incNumPartitions == zeroBasedPartition){                
+                    acceptedGlobalOrdinals.set(i);
+                }
+            }
```

Comment:
On the 2.7m userID terms test I see ords taking 2.4s vs 2.8s using hashes of terms.